syntax = "proto3";

service Analise {
    rpc AnalisarDadosPorTipo(DadosParaAnalise) returns (ResultadoAnalisePorTipo);
}

message DadosParaAnalise {
    string search_value = 1;
    string search_type = 2;  // "wavy", "tipo", or "both"
}

message TipoMedia {
    string tipo = 1;
    double media = 2;
    double min_valor = 3;
    double max_valor = 4;
    int32 total_amostras = 5;
}

message ResultadoAnalisePorTipo {
    repeated TipoMedia medias_por_tipo = 1;
}
